framework/utils/frontUtils.py

```diff
@@ -38,11 +38,21 @@ def nonDominatedFrontier(data, returnMask):
 
     @ In, data, np.array, data matrix (nPoints, nCosts) containing the data points
     @ In, returnMask, bool, type of data to be returned: indices (False) or True/False mask (True)
+    @ Out, minMask, np.array, array (nCosts,1) of boolean values: True (dimension need to be minimized), False (dimension need to be maximized)
     @ Out, isEfficientMask , np.array, data matrix (nPoints,1), array  of boolean values if returnMask=True
     @ Out, isEfficient, np.array, data matrix (nEfficientPoints,1), integer array of indexes if returnMask=False
 
     Reference: the following code has been adapted from https://stackoverflow.com/questions/32791911/fast-calculation-of-pareto-front-in-python
   """
+  if minMask is None:
+    pass
+  elif minMask is not None and minMask.shape[0] != data.shape[1]:
+    raise IOError("nonDominatedFrontier method: minMask has shape " + str(data.shape) + " while minMask has shape " + str(minMask.shape))
```

Comment:
Update the error message, it seems not clear to me

Reply:
message expanded and fixed